scripts/linters/pylint_extensions_test.py

```diff
@@ -3272,3 +3272,103 @@ def some_random_function(param1, param2):
 
         with self.checker_test_object.assertNoMessages():
             self.checker_test_object.checker.visit_functiondef(def_node)
+
+
+class StringConcatenationCheckerTests(unittest.TestCase):
+
+    def setUp(self):
+        super(StringConcatenationCheckerTests, self).setUp()
+        self.checker_test_object = testutils.CheckerTestCase()
+        self.checker_test_object.CHECKER_CLASS = (
+            pylint_extensions.StringConcatenationChecker)
+        self.checker_test_object.setup_method()
+
+    def test_string_concatenation_with_string_operands_raises_error(self):
+        binop_node1, binop_node2, binop_node3 = astroid.extract_node(
+            """
+            'string1' + 'string2' #@
+            a + 'string1' #@
+            'string2' + b #@
+            """)
+
+        str_str_message = testutils.Message(
+            msg_id='string-concatenation', args=('\'string1\''),
+            node=binop_node1)
+        right_message = testutils.Message(
+            msg_id='string-concatenation', args=('\'string1\''),
+            node=binop_node2)
+        left_message = testutils.Message(
+            msg_id='string-concatenation', args=('\'string2\''),
+            node=binop_node3)
+
+        with self.checker_test_object.assertAddsMessages(
+            str_str_message, right_message, left_message):
+            self.checker_test_object.checker.visit_binop(binop_node1)
+            self.checker_test_object.checker.visit_binop(binop_node2)
+            self.checker_test_object.checker.visit_binop(binop_node3)
```

Comment:
For this and other tests - please have a separate assert statement for every node (similar to rest of tests in this file) as it avoids any confusion i.e something like
```suggestion
        with self.checker_test_object.assertAddsMessages(
            str_str_message):
            self.checker_test_object.checker.visit_binop(binop_node1)

        with self.checker_test_object.assertAddsMessages(
            right_message):
            self.checker_test_object.checker.visit_binop(binop_node2)

        with self.checker_test_object.assertAddsMessages(
            left_message):
            self.checker_test_object.checker.visit_binop(binop_node3)     
```

Reply:
Done!!